### PR TITLE
Add Service Bindings support

### PR DESCRIPTION
**Pull request by sergey-melnychuk**
CloudFlare Workers Service Bindings:
https://developers.cloudflare.com/workers/platform/bindings/about-service-bindings/

Inspiration:
https://github.com/cloudflare/workers-rs/pull/183

This implementation was verified to work against live CloudFlare workers.

### CODE REVIEW

**Comment by sergey-melnychuk** [comment]
@zebp PTAL

File: worker-sys/src/service.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+use crate::Request;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen (extends = ::js_sys::Object, js_name = Fetcher)]
+    pub type Service;
```
Comment (zebp) [inline worker-sys/src/service.rs:7]: Now that dynamic dispatch is added, some of this boilerplate can be merged with [`Fetcher`](https://github.com/cloudflare/workers-rs/blob/main/worker-sys/src/fetcher.rs#L9).

File: worker/src/service.rs [hunk 2]
```diff
@@ -0,0 +1,58 @@
+use js_sys::Promise;
+use wasm_bindgen::{JsCast, JsValue};
+use wasm_bindgen_futures::JsFuture;
+
+use crate::{env::EnvBinding, Request, Response, Result};
+
+pub struct Service {
```
Comment (zebp) [inline worker/src/service.rs:7]: Same with this, should be able to merge this with the existing [`Fetcher`](https://github.com/cloudflare/workers-rs/blob/main/worker/src/fetcher.rs#L8).

**Review by zebp — CHANGES_REQUESTED** [review]
Thanks for the contribution, looks solid for the most part. A couple things like reusing existing dynamic dispatch binding logic and adding testing is needed so I'll be taking this PR over and finishing this up as this has become a priority.

**Comment by zebp** [comment]
Closing in favor of #243

**Pull request closed without merging**